Content.Shared/Body/Systems/SharedBloodstreamSystem.cs

```diff
@@ -289,25 +294,53 @@ private void OnRejuvenate(Entity<BloodstreamComponent> ent, ref RejuvenateEvent
     {
         TryModifyBleedAmount(ent.AsNullable(), -ent.Comp.BleedAmount);
 
-        if (SolutionContainer.ResolveSolution(ent.Owner, ent.Comp.BloodSolutionName, ref ent.Comp.BloodSolution, out var bloodSolution))
-            TryModifyBloodLevel(ent.AsNullable(), bloodSolution.AvailableVolume);
+        if (SolutionContainer.ResolveSolution(ent.Owner, ent.Comp.BloodSolutionName, ref ent.Comp.BloodSolution))
+        {
+            SolutionContainer.RemoveAllSolution(ent.Comp.BloodSolution.Value);
+            TryModifyBloodLevel(ent.AsNullable(), ent.Comp.BloodReferenceVolume);
+        }
+    }
+
+    private void OnMetabolismExclusion(Entity<BloodstreamComponent> ent, ref MetabolismExclusionEvent args)
+    {
+        // We explicitly want the exactly blood type that way consuming others blood makes us sick
+        var referenceBlood = new ReagentId(ent.Comp.BloodReagent, GetEntityBloodData(ent.Owner));
 
-        if (SolutionContainer.ResolveSolution(ent.Owner, ent.Comp.ChemicalSolutionName, ref ent.Comp.ChemicalSolution))
-            SolutionContainer.RemoveAllSolution(ent.Comp.ChemicalSolution.Value);
+        // Copy to list again to avoid modifying during enumeration!
+        foreach (var reagentQuantity in args.ReagentList.ToList())
+        {
+            if (reagentQuantity.Reagent != referenceBlood)
+                continue;
+
+            args.ReagentList.Remove(reagentQuantity);
+        }
```

Comment:
use [`List<T>.RemoveAll`](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.removeall?view=net-9.0) instead of duplicating the list